activation/poet.go

```diff
@@ -126,16 +129,42 @@ func WithLogger(logger *zap.Logger) PoetClientOpts {
 	}
 }
 
+func customLinearJitterBackoff(min, max time.Duration, attemptNum int, resp *http.Response) time.Duration {
+	attemptNum++
+	if max <= min {
+		return min * time.Duration(attemptNum)
+	}
+
+	source := rand.New(rand.NewSource(int64(time.Now().Nanosecond())))
+
+	jitter := source.Float64() * float64(max-min)
+	jitterMin := int64(jitter) + int64(min)
+
+	sleep := time.Duration(jitterMin * int64(attemptNum))
+	if sleep > max {
+		sleep = max
+	}
+	return sleep
+}
+
 // NewHTTPPoetClient returns new instance of HTTPPoetClient connecting to the specified url.
 func NewHTTPPoetClient(server types.PoetServer, cfg PoetConfig, opts ...PoetClientOpts) (*HTTPPoetClient, error) {
 	client := &retryablehttp.Client{
 		RetryMax:     cfg.MaxRequestRetries,
 		RetryWaitMin: cfg.RequestRetryDelay,
 		RetryWaitMax: 2 * cfg.RequestRetryDelay,
-		Backoff:      retryablehttp.LinearJitterBackoff,
+		Backoff:      customLinearJitterBackoff,
```

Comment:
Do we need our own custom implementation for the backoff?

Reply:
Also this isn't linear backoff - this is just retry at a constant rate 🙂

Reply:
With the current mainnet settings (max 10 retries) the retry behaviour will be:

Retry 1: wait randomly between 10 and 20 seconds
Retry 2-10: wait 20 seconds

And the node gives up latest after 200 seconds

while the previous behavior was that it would try 10 times: which take at least 550 seconds (always `min`) and at most 1100 seconds (always `max`) and aligns with the `RequestTimeout` of 1100 seconds.

I believe we don't need this custom implementation - with the default behavior the node will still give up after 10 retries which will take at least 550 seconds and at most 1100 seconds.